org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/SignatureHelpHandler.java

```diff
@@ -80,6 +88,37 @@ public SignatureHelp signatureHelp(TextDocumentPositionParams position, IProgres
 		return help;
 	}
 
+	private boolean isValid(ICompilationUnit unit, int offset, IProgressMonitor monitor) {
+		CompilationUnit ast = CoreASTProvider.getInstance().getAST(unit, CoreASTProvider.WAIT_YES, monitor);
+		boolean[] valid = { false };
+		ast.accept(new ASTVisitor() {
+
+			/* (non-Javadoc)
+			 * @see org.eclipse.jdt.core.dom.ASTVisitor#visit(org.eclipse.jdt.core.dom.MethodRef)
+			 */
+			@Override
+			public boolean visit(MethodRef node) {
+				if (node.getStartPosition() <= offset && (node.getStartPosition() + node.getLength()) >= offset) {
+					valid[0] = true;
+				}

```

Comment:
once we found something, we should stop visiting the AST

Reply:
Fixed.